asyncscope.md

```diff
@@ -450,17 +459,13 @@ namespace ex = std::execution;
 ex::sender auto some_work(int work_index);
 
 ex::sender auto foo(ex::scheduler auto sch) {
-    return unifex::let_value_with([]() noexcept { return ex::counting_scope{}; },
-                   [sch](ex::counting_scope& scope) {
+    return unifex::let_value_with_async_scope([sch](ex::counting_scope scope) {
```

Comment:
Two things: naming consistency, and the callable's argument is no longer a whole scope, but just a token, so maybe we update to `auto` for now?
```suggestion
    return unifex::let_with_async_scope([sch](auto scope) {
```